diff --git a/app/kdevelop.notifyrc b/app/kdevelop.notifyrc
index 1f040e7..5dc5782 100644
--- a/app/kdevelop.notifyrc
+++ b/app/kdevelop.notifyrc
@@ -471,7 +471,7 @@ Comment=A Breakpoint has been hit by the debugger
 Comment[bs]=Ispravljač grešaka je dosao do prijelomne tačke
 Comment[ca]=El depurador ha assolit un punt d'interrupció
 Comment[ca@valencia]=El depurador ha assolit un punt d'interrupció
-Comment[cs]=Bod přerušení byl dosažen debuggerem
+Comment[cs]=Bod přerušení byl dosažen ladicím nástrojem
 Comment[da]=Fejlsøgeren nåede et stoppunkt
 Comment[de]=Der Debugger hat einen Haltepunkt erreicht
 Comment[el]=Ο διορθωτής έχει σημειώσει προσωρινή παύση
@@ -544,7 +544,7 @@ Comment=A Watchpoint has been hit by the debugger
 Comment[bs]=Ispravljač grešaka je došao do pregledne tačke
 Comment[ca]=El depurador ha assolit un punt d'observació
 Comment[ca@valencia]=El depurador ha assolit un punt d'observació
-Comment[cs]=Bod sledování byl dosažen debuggerem
+Comment[cs]=Bod sledování byl dosažen ladicím nástrojem
 Comment[da]=Fejlsøgeren nåede et overvågningspunkt
 Comment[de]=Der Debugger hat einen Beobachtungspunkt erreicht
 Comment[el]=Ο διορθωτής έχει σημειώσει θέση επιθεώρησης
diff --git a/app_templates/cpp/CMake/cmake_qt4gui/cmake_qt4gui.kdevtemplate b/app_templates/cpp/CMake/cmake_qt4gui/cmake_qt4gui.kdevtemplate
index 7601ef2..7658185 100644
--- a/app_templates/cpp/CMake/cmake_qt4gui/cmake_qt4gui.kdevtemplate
+++ b/app_templates/cpp/CMake/cmake_qt4gui/cmake_qt4gui.kdevtemplate
@@ -52,7 +52,7 @@ Comment[it]=Applicazione GUI CMake in Qt4. Genera un'applicazione con GUI basata
 Comment[kk]=Qt4 CMake GUI бағдарламасы. CMake/Qt4 негіздеген графикалық интерфейсті бағдарламаны жасап береді (кроссплатформалық үлесімді)
 Comment[nb]=Qt4 Cmake Gui-program. Lag et program med grafisk brukerflate basert på CMake/Qt4 (kryssplattform-kompatibelt)
 Comment[nds]=Qt4-CMake-Böversietprogramm. En op CMake un Qt4 opbuut Böversiet-Programm opstellen (över Bedriefümgeven weg kompatibel)
-Comment[nl]=Gui-toepassing van Qt4 CMake. Genereert een ope CMake/Qt4 gebaseerd GUI-programma (crossplatform compatibel)
+Comment[nl]=Gui-toepassing van Qt4 CMake. Genereert een op CMake/Qt4 gebaseerd GUI-programma (crossplatform compatibel)
 Comment[pl]=Program Qt4 CMake Gui. Generuj programy oparte o CMake/Qt4 (kompatybilne na wielu platformach)
 Comment[pt]=Aplicação gráfica do Qt4 com CMake. Gera uma aplicação gráfica baseada no CMake/Qt4 (compatível com várias plataformas)
 Comment[pt_BR]=Aplicativo gráfico com CMake para Qt4. Gera um aplicativo de interface em CMake/Qt4 (compatível com várias plataformas)
diff --git a/debuggers/CMakeLists.txt b/debuggers/CMakeLists.txt
index 8fe222c..592c1fc 100644
--- a/debuggers/CMakeLists.txt
+++ b/debuggers/CMakeLists.txt
@@ -1,4 +1,4 @@
-if(NOT WIN32 AND NOT APPLE)
+if(NOT WIN32)
   add_subdirectory(gdb)
-endif(NOT WIN32 AND NOT APPLE )
+endif(NOT WIN32)
 
diff --git a/debuggers/gdb/CMakeLists.txt b/debuggers/gdb/CMakeLists.txt
index 3d1125c..ef6100f 100644
--- a/debuggers/gdb/CMakeLists.txt
+++ b/debuggers/gdb/CMakeLists.txt
@@ -1,11 +1,13 @@
 project(gdb)
 
+if(NOT APPLE)
 macro_optional_find_package( KDE4Workspace 4.2.0 )
 
 macro_log_feature( KDE4WORKSPACE_FOUND
 "KDE 4 Workspace" "KDE 4 Workspace libraries" "kdebase/workspace" FALSE
 "" "Required for building the process selection dialog for attaching GDB to a running process."
 )
+endif(NOT APPLE)
 
 if(KDE4WORKSPACE_FOUND)
     include_directories(
@@ -80,7 +82,7 @@ set(kdevgdb_PART_UI
 
 kde4_add_ui_files(kdevgdb_PART_SRCS ${kdevgdb_PART_UI})
 kde4_add_plugin(kdevgdb ${kdevgdb_PART_SRCS})
-target_link_libraries(kdevgdb sublime
+target_link_libraries(kdevgdb ${KDEVPLATFORM_SUBLIME_LIBRARIES}
     ${KDEVPLATFORM_INTERFACES_LIBRARIES}
     ${KDEVPLATFORM_LANGUAGE_LIBRARIES}
     ${KDEVPLATFORM_DEBUGGER_LIBRARIES}
diff --git a/debuggers/gdb/gdb.cpp b/debuggers/gdb/gdb.cpp
index 68e1768..4642087 100644
--- a/debuggers/gdb/gdb.cpp
+++ b/debuggers/gdb/gdb.cpp
@@ -44,7 +44,7 @@
 using namespace GDBDebugger;
 
 GDB::GDB(QObject* parent)
-: QObject(parent), process_(0), sawPrompt_(false), currentCmd_(0), receivedReply_(false), isRunning_(false)
+: QObject(parent), process_(0), sawPrompt_(false), currentCmd_(0), receivedReply_(false), isRunning_(false), childPid_(0)
 {
 }
 
@@ -157,8 +157,11 @@ void GDB::interrupt()
 {
     //TODO:win32 Porting needed
     int pid = process_->pid();
-    if (pid != 0)
+    if (childPid_) {
+        ::kill(childPid_, SIGINT);
+    } else if (pid != 0) {
         ::kill(pid, SIGINT);
+    }
 }
 
 GDBCommand* GDB::currentCommand() const
@@ -250,7 +253,24 @@ void GDB::processLine(const QByteArray& line)
                GDBMI::ResultRecord& result = static_cast<GDBMI::ResultRecord&>(*r);
 
                emit internalCommandOutput(QString::fromUtf8(line) + '\n');
-               
+
+               if (result.reason == "thread-group-started") {
+                   receivedReply_ = true;
+                   //     (gdb) -exec-run
+                   //     =thread-group-started,id="i1",pid="16768"
+                   if (line.contains("pid=\"")) {
+                       QList<QByteArray> splitLine = line.split(',');
+                       if (splitLine.size() > 2) {
+                           QByteArray pidStr = splitLine[2];
+                           pidStr.chop(1);
+                           childPid_ = pidStr.mid(5).toULong();
+                           if (childPid_ == 0) {
+                               kDebug() << "line=" << line << "pidStr=" << pidStr << pidStr.mid(5) << "pid=" << childPid_;
+                           }
+                       }
+                   }
+               }
+
                // FIXME: the code below should be reviewed to consider result record
                // subtype when doing all decisions.
                
diff --git a/debuggers/gdb/gdb.h b/debuggers/gdb/gdb.h
index 6f99a60..31688cd 100644
--- a/debuggers/gdb/gdb.h
+++ b/debuggers/gdb/gdb.h
@@ -144,6 +144,7 @@ private:
     
     bool receivedReply_;
     bool isRunning_;
+    unsigned long childPid_;
 };
 }
 
diff --git a/debuggers/gdb/stty.cpp b/debuggers/gdb/stty.cpp
index 736fff4..606977b 100644
--- a/debuggers/gdb/stty.cpp
+++ b/debuggers/gdb/stty.cpp
@@ -160,14 +160,19 @@ int STTY::findTTY()
 
 #ifdef __sgi__
     ptyfd = open("/dev/ptmx",O_RDWR);
-    if (ptyfd < 0) {
+#elif defined(Q_OS_MAC)
+    ptyfd = posix_openpt(O_RDWR);
+#endif
+#if defined(__sgi__) || defined(Q_OS_MAC)
+    if (ptyfd == -1) {
         perror("Can't open a pseudo teletype");
         return(-1);
+    } else if (ptyfd >= 0) {
+        strncpy(tty_slave, ptsname(ptyfd), 50);
+        grantpt(ptyfd);
+        unlockpt(ptyfd);
+        needGrantPty = false;
     }
-    strncpy(tty_slave, ptsname(ptyfd), 50);
-    grantpt(ptyfd);
-    unlockpt(ptyfd);
-    needGrantPty = false;
 #endif
 
     // first we try UNIX PTY's
diff --git a/file_templates/classes/qobject/qobject.desktop b/file_templates/classes/qobject/qobject.desktop
index 6e92967..aaade37 100644
--- a/file_templates/classes/qobject/qobject.desktop
+++ b/file_templates/classes/qobject/qobject.desktop
@@ -13,7 +13,7 @@ Name[fi]=QObject-aliluokka
 Name[fr]=Sous-classe QObject
 Name[gl]=Subclase de QObject
 Name[hu]=QObject alosztály
-Name[it]=Sottoclasse GObject
+Name[it]=Sottoclasse QObject
 Name[kk]=QObject субкласы
 Name[nb]=QObject subklasse
 Name[nds]=QObject-Ünnerklass
@@ -43,7 +43,7 @@ Comment[fi]=QObject-aliluokka ominaisuuksilla
 Comment[fr]=Sous-classe QObject avec des propriétés
 Comment[gl]=Subclase de QObject con propiedades
 Comment[hu]=QObject alosztály tulajdonságokkal
-Comment[it]=Sottoclasse GObject con delle proprietà
+Comment[it]=Sottoclasse QObject con delle proprietà
 Comment[kk]=Қасиеттері бар QObject субкласы
 Comment[nb]=QObject subklasse med egenskaper
 Comment[nds]=QObject-Ünnerklass mit Egenschappen
diff --git a/languages/cpp/codegen/adaptsignatureassistant.cpp b/languages/cpp/codegen/adaptsignatureassistant.cpp
index 087c0ed..5b13410 100644
--- a/languages/cpp/codegen/adaptsignatureassistant.cpp
+++ b/languages/cpp/codegen/adaptsignatureassistant.cpp
@@ -94,7 +94,7 @@ void AdaptSignatureAssistant::reset()
   m_otherSideId = {};
   m_otherSideTopContext = {};
   m_otherSideContext = {};
-  m_oldSignature = {};
+  m_oldSignature = Signature();
   m_document = {};
   m_view.clear();
 }
diff --git a/languages/cpp/includepathcomputer.cpp b/languages/cpp/includepathcomputer.cpp
index 61c3646..323577a 100644
--- a/languages/cpp/includepathcomputer.cpp
+++ b/languages/cpp/includepathcomputer.cpp
@@ -111,7 +111,7 @@ void IncludePathComputer::computeForeground()
       m_effectiveBuildDirectory.clear();
     }
 
-      for (const auto& dir : idm->includes(file, IDefinesAndIncludesManager::Type(IDefinesAndIncludesManager::UserDefined | IDefinesAndIncludesManager::CompilerSpecific))){
+      for (const auto& dir : idm->includes(file, IDefinesAndIncludesManager::Type(IDefinesAndIncludesManager::UserDefined))){
         addInclude(dir);
       }
       m_defines = idm->defines(file);
@@ -121,6 +121,10 @@ void IncludePathComputer::computeForeground()
     foreach (const Path& dir, dirs) {
       addInclude(dir);
     }
+
+    for (const auto& dir : idm->includes(file, IDefinesAndIncludesManager::Type(IDefinesAndIncludesManager::CompilerSpecific))){
+      addInclude(dir);
+    }
   }
 
   if (noProject) {
diff --git a/languages/cpp/parser/lexer.cpp b/languages/cpp/parser/lexer.cpp
index ace64b7..2b6a843 100644
--- a/languages/cpp/parser/lexer.cpp
+++ b/languages/cpp/parser/lexer.cpp
@@ -313,9 +313,7 @@ void Lexer::tokenize(ParseSession* _session)
     size_t previousIndex = index;
 
     {
-    Token token;
-    token.position = cursor.offsetIn( session->contents() );
-    token.size = 0;
+    Token token{cursor.offsetIn(session->contents()), 0, Token_EOF};
     stream->append(token);
     }
     Token* current_token = &(session->token_stream->last());
diff --git a/languages/plugins/custom-definesandincludes/compilerprovider/compilerprovider.cpp b/languages/plugins/custom-definesandincludes/compilerprovider/compilerprovider.cpp
index 6c53e57..5d5b5fa 100644
--- a/languages/plugins/custom-definesandincludes/compilerprovider/compilerprovider.cpp
+++ b/languages/plugins/custom-definesandincludes/compilerprovider/compilerprovider.cpp
@@ -167,6 +167,9 @@ CompilerProvider::CompilerProvider( QObject* parent, const QVariantList& )
     : IPlugin( CompilerProviderFactory::componentData(), parent )
 {
     KDEV_USE_EXTENSION_INTERFACE( ICompilerProvider );
+    /// FIXME: cleanup the code base here, don't make the provider a plugin but a static library or similar.
+    ///        simplify the code in generel, less interfaces/virtuals!
+    static_cast<DefinesAndIncludesManager*>( IDefinesAndIncludesManager::manager() )->setProvider(this);
 
     m_factories.append(CompilerFactoryPointer(new GccFactory()));
     m_factories.append(CompilerFactoryPointer(new ClangFactory()));
@@ -198,6 +201,9 @@ CompilerProvider::CompilerProvider( QObject* parent, const QVariantList& )
 
     //Add a provider for files without project
     addPoject( nullptr, checkCompilerExists({}));
+    for (auto project : ICore::self()->projectController()->projects()) {
+        projectOpened( project );
+    }
 }
 
 QVector< CompilerPointer > CompilerProvider::compilers() const
diff --git a/languages/plugins/custom-definesandincludes/compilerprovider/kdevcompilerprovider.desktop.cmake b/languages/plugins/custom-definesandincludes/compilerprovider/kdevcompilerprovider.desktop.cmake
index 8d19603..fd1e020 100644
--- a/languages/plugins/custom-definesandincludes/compilerprovider/kdevcompilerprovider.desktop.cmake
+++ b/languages/plugins/custom-definesandincludes/compilerprovider/kdevcompilerprovider.desktop.cmake
@@ -8,6 +8,8 @@ Name[en_GB]=Compiler Provider
 Name[es]=Proveedor del compilador
 Name[et]=Kompilaatori pakkuja
 Name[fi]=Käännintarjoaja
+Name[fr]=Fournisseur de compilateur
+Name[it]=Fornitore compilatore
 Name[nl]=Leverancier van compiler
 Name[pl]=Dostawca kompilatora
 Name[pt]=Fornecedor de Compiladores
@@ -25,6 +27,8 @@ GenericName[en_GB]=Compiler Provider
 GenericName[es]=Proveedor del compilador
 GenericName[et]=Kompilaatori pakkuja
 GenericName[fi]=Käännintarjoaja
+GenericName[fr]=Fournisseur de compilateur
+GenericName[it]=Fornitore compilatore
 GenericName[nl]=Leverancier van compiler
 GenericName[pl]=Dostawca kompilatora
 GenericName[pt]=Fornecedor de Compiladores
diff --git a/languages/plugins/custom-definesandincludes/kcm_widget/kcm_kdevcustomdefinesandincludes.desktop b/languages/plugins/custom-definesandincludes/kcm_widget/kcm_kdevcustomdefinesandincludes.desktop
index d88a4b6..a4724bf 100644
--- a/languages/plugins/custom-definesandincludes/kcm_widget/kcm_kdevcustomdefinesandincludes.desktop
+++ b/languages/plugins/custom-definesandincludes/kcm_widget/kcm_kdevcustomdefinesandincludes.desktop
@@ -15,6 +15,8 @@ Name[en_GB]=Custom Defines And Includes
 Name[es]=Personalizar definiciones e inclusiones
 Name[et]=Kohandatud definitsioonid ja päised
 Name[fi]=Mukautetut definet ja includet
+Name[fr]=Définitions et inclusions personnalisées
+Name[it]=Definizioni e inclusioni personalizzate
 Name[nl]=Zelf gedefinieerd en invoegingen
 Name[pl]=Własne definicje i pliki dołączane
 Name[pt]=Definições e Inclusões Personalizadas
@@ -32,6 +34,8 @@ Comment[en_GB]=Configure which macros and include directories/files will be adde
 Comment[es]=Configurar las macros y los directorios/archivos a incluir que se añadirán al analizador sintáctico durante el análisis de los proyectos.
 Comment[et]=Aitab seadistada, millised makrod ja päistekataloogid või -failid lisada parserile projekti parsimiseks.
 Comment[fi]=Määritä, mitkä makrot ja include-hakemistot/tiedostot lisätään jäsentimelle projektin jäsentämisen aikana.
+Comment[fr]=Configurer quelles macros et dossiers ou fichiers d'inclusions seront ajoutés à l'analyseur pendant l'analyse du projet.
+Comment[it]=Configura quali macro e cartelle/file di inclusione saranno aggiunti all'analizzatore durante l'analisi del progetto.
 Comment[nl]=Stel in welke macro's en ingevoegde mappen/bestanden toegevoegd zullen worden aan de parser bij het ontleden van het project.
 Comment[pl]=Ustaw jakie pliki/katalogi makr i plików dołączanych będą dodawane do przetwarzania przy przetwarzaniu projektu.
 Comment[pt]=Configure quais as macros e pastas/ficheiros de inclusão que serão adicionados ao processador do projecto.
diff --git a/languages/plugins/custom-definesandincludes/kdevdefinesandincludesmanager.desktop.cmake b/languages/plugins/custom-definesandincludes/kdevdefinesandincludesmanager.desktop.cmake
index 57f845d..a5bbc1c 100644
--- a/languages/plugins/custom-definesandincludes/kdevdefinesandincludesmanager.desktop.cmake
+++ b/languages/plugins/custom-definesandincludes/kdevdefinesandincludesmanager.desktop.cmake
@@ -7,6 +7,8 @@ Name[en_GB]=Custom Defines And Includes Manager
 Name[es]=Gestor de definiciones e inclusiones personalizadas
 Name[et]=Kohandatud definitsioonide ja päiste haldur
 Name[fi]=Mukautettujen definejen ja includejen hallinta
+Name[fr]=Gestionnaire de définitions et inclusions personnalisées
+Name[it]=Gestore delle definizioni e inclusioni personalizzate
 Name[nl]=Beheerder van zelf gedefinieerd en invoegingen
 Name[pl]=Zarządzanie własnymi definicjami i plikami dołączanymi
 Name[pt]=Gestor de Definições e Inclusões Personalizadas
@@ -23,6 +25,8 @@ GenericName[en_GB]=Custom Defines And Includes Manager
 GenericName[es]=Gestor de definiciones e inclusiones personalizadas
 GenericName[et]=Kohandatud definitsioonide ja päiste haldur
 GenericName[fi]=Mukautettujen definejen ja includejen hallinta
+GenericName[fr]=Gestionnaire de définitions et inclusions personnalisées
+GenericName[it]=Gestore delle definizioni e inclusioni personalizzate
 GenericName[nl]=Beheerder van zelf gedefinieerd en invoegingen
 GenericName[pl]=Zarządzanie własnymi definicjami i plikami dołączanymi
 GenericName[pt]=Gestor de Definições e Inclusões Personalizadas
@@ -39,6 +43,8 @@ Comment[en_GB]=Configure which macros and include directories/files will be adde
 Comment[es]=Configurar las macros y los directorios/archivos a incluir que se añadirán al analizador sintáctico durante el análisis de los proyectos.
 Comment[et]=Aitab seadistada, millised makrod ja päistekataloogid või -failid lisada parserile projekti parsimiseks.
 Comment[fi]=Määritä, mitkä makrot ja include-hakemistot/tiedostot lisätään jäsentimelle projektin jäsentämisen aikana.
+Comment[fr]=Configurer quelles macros et dossiers ou fichiers d'inclusions seront ajoutés à l'analyseur pendant l'analyse du projet.
+Comment[it]=Configura quali macro e cartelle/file di inclusione saranno aggiunti all'analizzatore durante l'analisi del progetto.
 Comment[nl]=Stel in welke macro's en ingevoegde mappen/bestanden toegevoegd zullen worden aan de parser bij het ontleden van het project.
 Comment[pl]=Ustaw jakie pliki/katalogi makr i plików dołączanych będą dodawane do przetwarzania przy przetwarzaniu projektu.
 Comment[pt]=Configure quais as macros e pastas/ficheiros de inclusão que serão adicionados ao processador do projecto.
diff --git a/languages/plugins/custom-definesandincludes/settingsmanager.cpp b/languages/plugins/custom-definesandincludes/settingsmanager.cpp
index 9e09798..fb31f39 100644
--- a/languages/plugins/custom-definesandincludes/settingsmanager.cpp
+++ b/languages/plugins/custom-definesandincludes/settingsmanager.cpp
@@ -51,9 +51,15 @@ const QString compilerTypeKey = QLatin1String( "Type" );
 }
 
 SettingsManager::SettingsManager()
+    : m_provider(nullptr)
 {
 }
 
+void SettingsManager::setProvider(const ICompilerProvider* provider)
+{
+    m_provider = provider;
+}
+
 namespace
 {
 void doWriteSettings( KConfigGroup grp, const QList<ConfigEntry>& paths )
@@ -135,15 +141,6 @@ QList<ConfigEntry> convertedPaths( KConfig* cfg )
     return paths;
 }
 
-ICompilerProvider* currentCompilerProvider()
-{
-    auto compilerProvider = KDevelop::ICore::self()->pluginController()->pluginForExtension("org.kdevelop.ICompilerProvider");
-    if (!compilerProvider || !compilerProvider->extension<ICompilerProvider>()) {
-        return {};
-    }
-
-    return compilerProvider->extension<ICompilerProvider>();
-}
 }
 
 void SettingsManager::writePaths( KConfig* cfg, const QList< ConfigEntry >& paths )
@@ -175,13 +172,15 @@ QList<ConfigEntry> SettingsManager::readPaths( KConfig* cfg ) const
 
 CompilerPointer SettingsManager::currentCompiler( KConfig* cfg, const CompilerPointer& defaultCompiler ) const
 {
+    Q_ASSERT(m_provider);
+
     auto grp = cfg->group( ConfigConstants::definesAndIncludesGroup ).group("Compiler");
     auto name = grp.readEntry( ConfigConstants::compilerNameKey, QString() );
     if (name.isEmpty()) {
         return {};
     }
 
-    for (auto c : currentCompilerProvider()->compilers()) {
+    for (auto c : m_provider->compilers()) {
         if (c->name() == name) {
             return c;
         }
@@ -190,7 +189,7 @@ CompilerPointer SettingsManager::currentCompiler( KConfig* cfg, const CompilerPo
     auto path = grp.readEntry( ConfigConstants::compilerPathKey, QString() );
     auto type = grp.readEntry( ConfigConstants::compilerTypeKey, QString() );
 
-    auto cf = currentCompilerProvider()->compilerFactories();
+    auto cf = m_provider->compilerFactories();
     for (auto f : cf) {
         if (f->name() == type) {
             return f->createCompiler(name, path, true);
@@ -241,6 +240,8 @@ void SettingsManager::writeUserDefinedCompilers(const QVector< CompilerPointer >
 
 QVector< CompilerPointer > SettingsManager::userDefinedCompilers() const
 {
+    Q_ASSERT(m_provider);
+
     QVector< CompilerPointer > compilers;
 
     KConfigGroup config = KGlobal::config()->group(ConfigConstants::compilersGroup);
@@ -252,7 +253,7 @@ QVector< CompilerPointer > SettingsManager::userDefinedCompilers() const
         auto path = grp.readEntry(ConfigConstants::compilerPathKey, QString());
         auto type = grp.readEntry(ConfigConstants::compilerTypeKey, QString());
 
-        auto cf = currentCompilerProvider()->compilerFactories();
+        auto cf = m_provider->compilerFactories();
         for (auto f : cf) {
             if (f->name() == type) {
                 compilers.append(f->createCompiler(name, path));
diff --git a/languages/plugins/custom-definesandincludes/settingsmanager.h b/languages/plugins/custom-definesandincludes/settingsmanager.h
index a329879..23c0d10 100644
--- a/languages/plugins/custom-definesandincludes/settingsmanager.h
+++ b/languages/plugins/custom-definesandincludes/settingsmanager.h
@@ -34,10 +34,14 @@ namespace KDevelop
 class IProject;
 }
 
+class ICompilerProvider;
 class KDEVDEFINESANDINCLUDESMANAGER_EXPORT SettingsManager
 {
 public:
     SettingsManager();
+    ~SettingsManager(){}
+
+    void setProvider(const ICompilerProvider* provider);
 
     QList<KDevelop::ConfigEntry> readPaths(KConfig* cfg) const;
 
@@ -55,7 +59,8 @@ public:
 
     bool needToReparseCurrentProject( KConfig* cfg ) const;
 
-    ~SettingsManager(){}
+private:
+    const ICompilerProvider* m_provider;
 };
 
 #endif // SETTINGSMANAGER_H
diff --git a/projectmanagers/cmake/settings/cmakepreferences.cpp b/projectmanagers/cmake/settings/cmakepreferences.cpp
index 6db6974..8b91cfd 100644
--- a/projectmanagers/cmake/settings/cmakepreferences.cpp
+++ b/projectmanagers/cmake/settings/cmakepreferences.cpp
@@ -82,7 +82,7 @@ CMakePreferences::CMakePreferences(QWidget* parent, const QVariantList& args)
     connect(m_prefsUi->addBuildDir, SIGNAL(pressed()), this, SLOT(createBuildDir()));
     connect(m_prefsUi->removeBuildDir, SIGNAL(pressed()), this, SLOT(removeBuildDir()));
     connect(m_prefsUi->showAdvanced, SIGNAL(toggled(bool)), this, SLOT(showAdvanced(bool)));
-    connect(m_prefsUi->environment, SIGNAL(currentIndexChanged(int)), this, SLOT(changed()));
+    connect(m_prefsUi->environment, SIGNAL(currentProfileChanged(QString)), this, SLOT(changed()));
     
     showInternal(m_prefsUi->showInternal->checkState());
     m_subprojFolder=KUrl(args[1].toString()).upUrl();
diff --git a/providers/ghprovider/kdevghprovider.desktop.cmake b/providers/ghprovider/kdevghprovider.desktop.cmake
index 619ff35..c3076c7 100644
--- a/providers/ghprovider/kdevghprovider.desktop.cmake
+++ b/providers/ghprovider/kdevghprovider.desktop.cmake
@@ -33,7 +33,7 @@ Name=Github Provider
 Name[bs]=Github pružalac
 Name[ca]=Proveïdor de Github
 Name[ca@valencia]=Proveïdor de Github
-Name[cs]=Poskytovatel Github
+Name[cs]=Poskytovatel GitHubu
 Name[da]=Github-udbyder
 Name[de]=GitHub-Provider
 Name[el]=Πάροχος Github
@@ -62,7 +62,7 @@ GenericName=Github Provider
 GenericName[bs]=Github pružalac
 GenericName[ca]=Proveïdor de Github
 GenericName[ca@valencia]=Proveïdor de Github
-GenericName[cs]=Poskytovatel Github
+GenericName[cs]=Poskytovatel GitHubu
 GenericName[da]=Github-udbyder
 GenericName[de]=GitHub-Provider
 GenericName[el]=Πάροχος Github
