diff --git a/languages/cpp/codegen/ui/custom_include_paths.ui b/languages/cpp/codegen/ui/custom_include_paths.ui
--- a/languages/cpp/codegen/ui/custom_include_paths.ui
+++ b/languages/cpp/codegen/ui/custom_include_paths.ui
@@ -7,7 +7,7 @@
     <x>0</x>
     <y>0</y>
     <width>526</width>
-    <height>504</height>
+    <height>536</height>
    </rect>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
@@ -113,14 +113,8 @@
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label_2">
-        <property name="sizePolicy">
-         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
-          <horstretch>0</horstretch>
-          <verstretch>0</verstretch>
-         </sizepolicy>
-        </property>
         <property name="text">
-         <string>You can add a list of custom include paths here.</string>
+         <string>You can add a list of custom include paths here. Relative paths will be resolved from the storage directory.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
diff --git a/languages/cpp/includepathresolver.cpp b/languages/cpp/includepathresolver.cpp
--- a/languages/cpp/includepathresolver.cpp
+++ b/languages/cpp/includepathresolver.cpp
@@ -381,6 +381,9 @@ CustomIncludePathsSettings CustomIncludePathsSettings::read(QString storagePath)
               }
             }
           }else{
+            // Turn relative paths into absolute paths from the storage path
+            if (!textLine.startsWith("/"))
+              textLine = sourceDir.absoluteFilePath(textLine);
             ret.paths << textLine;
           }
         }
