--- kdevelop-4.5.1/debuggers/gdb/printers/gdbinit.old	2013-05-29 23:01:36.000000000 +0200
+++ kdevelop-4.5.1/debuggers/gdb/printers/gdbinit	2013-10-26 20:09:13.220314577 +0200
@@ -8,7 +8,4 @@
 from kde4 import register_kde4_printers
 register_kde4_printers (None)
 
-from libstdcxx import register_libstdcxx_printers
-register_libstdcxx_printers (None)
-
 end
--- kdevelop-4.5.1/debuggers/gdb/printers/CMakeLists.txt.old	2013-05-29 23:01:36.000000000 +0200
+++ kdevelop-4.5.1/debuggers/gdb/printers/CMakeLists.txt	2013-10-26 20:08:52.916980787 +0200
@@ -2,6 +2,6 @@
 
 configure_file( gdbinit ${CMAKE_CURRENT_BINARY_DIR}/gdbinit )
 
-install( FILES qt4.py kde4.py libstdcxx.py
+install( FILES qt4.py kde4.py
     ${CMAKE_CURRENT_BINARY_DIR}/gdbinit
     DESTINATION ${DATA_INSTALL_DIR}/kdevgdb/printers )
